.github/workflows/build-test-distribute.yaml

```
@@ -199,6 +388,72 @@ jobs:
           GITHUB_APP: "true"
           GH_TOKEN: ${{ steps.github-app-token.outputs.token }}
         run: make helm/release
+  # The provenance job is reusable workflow that:
+  # Cannot be used as a step within other jobs
+  # Cannot be used to pass ENV variables
+  # GHA limitations don't allow dynamic output keys / expressions
+  provenance-metadata:
+    name: Provenance metadata
+    needs: [metadata, distributions, distribute-images, sign-image-manifests]
+    runs-on: ubuntu-latest-kong
+    if: >
+      github.repository_owner == 'Kong'  && needs.metadata.outputs.ALLOW_PUSH == 'true' && github.event_name != 'pull_request'
+
+    outputs:
+      artifact-provenance-result: ${{ needs.distributions.outputs.subjects-as-file }}
+      image-manifest-provenance-result: ${{ steps.image-manifest-provenance-metadata.outputs.result }}
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          fetch-depth: 0
+      - uses: cloudposse/github-action-matrix-outputs-read@5690ce6664e0dcdf06e27e4ba27e7a424951da56
+        id: image-manifest-provenance-metadata
+        with:
+          matrix-step-name: sign-image-manifests # Must match the job name where the matrix-outputs-write action is invoked
+      - run: |
+          echo ${{ steps.image-manifest-provenance-metadata.outputs.result }}
+  # Provenance job for all images manifests
+  # SLSA generator is a reusable workflow
+  # pull-request event is [not supported](https://github.com/slsa-framework/slsa-github-generator/tree/main/internal/builders/container#supported-triggers)
+  # runs-on option is [not supported](https://github.com/orgs/community/discussions/25783)
+  # ENV option is [not supported](https://github.com/orgs/community/discussions/26671)
+  # Reusable workflow doesn't support exrernal COSIGN_REPOSITORY via input / env variable
+  # TODO:
+  #   Split provenance jobs for internal / official releases when repositories are split
+  images-provenance:
+    name: Images Provenance
+    needs: [metadata, distribute-images, sign-image-manifests, provenance-metadata]
+    if: >
+      github.repository_owner == 'Kong' && github.event_name != 'pull_request' && needs.metadata.outputs.ALLOW_PUSH == 'true'
```

```suggestion
      needs.metadata.outputs.ALLOW_PUSH == 'true'
``` 

This should be good enough no?

---

`needs.metadata.outputs.ALLOW_PUSH == 'true' and github.event_name != 'pull_request'` is needed due to the [limitation](https://github.com/slsa-framework/slsa-github-generator/tree/main/internal/builders/container#supported-triggers) mentione here for supported triggers

---

Does `needs.metadata.outputs.ALLOW_PUSH == 'true'` guarantee to run this job only for `kumahq` owned repos?

---

done